cardGameAll.xml

```diff
@@ -2034,76 +2040,54 @@
-<Command>		if (HasKeyword(GUARD, 1*yGetVar(db, "keywords"))) {</Command>
-<Command>			tileGuard(tile, false);</Command>
-<Command>		}</Command>
-
-<Command>		int n = 1*xsPow(2, DEATH_EVENT_COUNT - 1);</Command>
-<Command>		int events = 1*yGetVar(db, "OnDeath");</Command>
-<Command>		for(x=DEATH_EVENT_COUNT - 1; >=0) {</Command>
-<Command>			if (events >= n) {</Command>
-<Command>				if(OnDeath(index, x)){</Command>
-<Command>					checkAgain = true;</Command>
-<Command>				}</Command>
-<Command>				events = events - n;</Command>
-<Command>			}</Command>
-<Command>		n = n / 2;</Command>
-<Command>		}</Command>
-
-<Command>		removeUnit(db);</Command>
-<Command>	}</Command>
-<Command>	return (checkAgain);</Command>
-<Command>}</Command>
-
-
-<Command>rule DelayedDeathCheck</Command>
-<Command>highFrequency</Command>
-<Command>inactive</Command>
-<Command>{</Command>
-<Command>	if ((trTime()-cActivationTime) > 0){</Command>
-<Command>		trChatSend(0, "debug:I am in DelayedDeathCheck");</Command>
-<Command>		bool finalCheck = true;</Command>
+<Command>void removeDeadUnits() {</Command>
+<Command>	int savePointer = yGetPointer("allUnits");</Command>
+<Command>	bool checkAgain = true;</Command>
+<Command>	while(checkAgain){</Command>
+<Command>		checkAgain = false;</Command>
 <Command>		yDatabasePointerDefault("allUnits");</Command>
 <Command>		for(y=yGetDatabaseCount("allUnits"); >0) {</Command>
 <Command>			yDatabaseNext("allUnits");</Command>
-<Command>			if(removeIfDead("allUnits")){</Command>
-<Command>				finalCheck = false;</Command>
+<Command><![CDATA[			if (yGetVar("allUnits", "health") <= 0 && yGetVar("allUnits", "OnDeath") > 0) {]]></Command>
+<Command>				int events = 1*yGetVar("allUnits", "OnDeath");</Command>
+<Command>				ySetVar("allUnits", "OnDeath", 0);</Command>
+<Command>				int n = 1*xsPow(2, DEATH_EVENT_COUNT - 1);</Command>
+<Command>				for(x=DEATH_EVENT_COUNT - 1; >=0) {</Command>
+<Command>					if (events >= n) {</Command>
+<Command>						if(OnDeath(x)){</Command>
+<Command>							checkAgain = true;</Command>
+<Command>						}</Command>
+<Command>						events = events - n;</Command>
+<Command>					}</Command>
+<Command>					n = n / 2;</Command>
+<Command>				}</Command>
 <Command>			}</Command>
 <Command>		}</Command>
-<Command>		if(finalCheck){</Command>
-<Command>			xsEnableRule("gameplay_01_select");</Command>
-<Command>			highlightReady(100);</Command>
+<Command>	}</Command>
+<Command>	yDatabasePointerDefault("allUnits");</Command>
+<Command>	for(y=yGetDatabaseCount("allUnits"); >0) {</Command>
+<Command>		yDatabaseNext("allUnits");</Command>
+<Command><![CDATA[		if (yGetVar("allUnits", "health") <= 0 && yGetVar("allUnits", "health") > -9000) {]]></Command>
+<Command>			int tile = yGetVar("allUnits", "tile");</Command>
+<Command>			zSetVarByIndex("tiles", "occupied", tile, xsMax(TILE_EMPTY, zGetVarByIndex("tiles", "terrain", tile)));</Command>
+<Command>			trDamageUnitPercent(100);</Command>
```

Comment:
there should be trUnitSelectClear(); and trUnitSelect(""+1*trQuestVarGet("allunits"), true);

Reply:
or, on line 2078, make yDatabaseNext("allUnits", true);

yDatabaseNext() prefers not to select the unit because of the searching thingy. have to manually set it to select the unit